### ISSUE TITLE
Stick to button alignment does not account for container position/margin/padding

### ISSUE THREAD

**Issue by brickers** [mendixlabs/listview-swipe#3]
I have a project where I have a list view within a container for UI purposes - we do not wish the list view items to take up the full width of the screen. Each list view has one swipe item on the right, with the widget set to 'stick to button'.

When the left side of the container is moved away from the edge of the screen, either by positioning, or adding margins or padding, the foreground of the widget overlaps the background button by the distance the side was moved.

This can be accounted for manually by reducing the width of the foreground when it has the class .swiping-left by that distance. However this is not practical once auto margins come into play. Any ideas how this could be resolved?

**Comment by Andries-Smit** [comment]
Dear Feargus,

We have not encounter a similar challenge. The background is moved in the dom. And is positioned absolute to its parent (dataview) and does not care about the margins of the foreground.

Margin style should be applied to the dataview
``` css
.mx-listview-item > .mx-dataview {
    margin-left: 20px;
    margin-right: 20px;
}
```
Add it to the sample https://listviewswipeout.mxapps.io/index-phone-preview.html?profile=phone

This way; the single margin will be applied for both foreground and background. This would probably also help with your auto margins.

If that does not solve your issue, please provide us with a sample project, to see if i can help you out, else  I will put this question forward to our UX team.

Cheers,
Andries

**Comment by brickers** [comment]
Hi Andries

Thanks for looking so quickly, and over the weekend too. I'm not looking to provide margins or padding at that level. Looking at the sample you linked to, please add margin to the div.mx-listview that is the direct parent of the ul.mx-list. You will see that on both sides the foreground is misplaced relative to the buttons by the amount of margin (please see attached screenshot). I believe this will occur with any movement of the left side at any element higher up the hierarchy. 

![swipe-widget-error](https://user-images.githubusercontent.com/8740034/38485719-b23e7740-3bd2-11e8-8853-3a1dbffb2ab4.png)

Thanks
Ferg

**Comment by Andries-Smit** [comment]
Fixed in version 1.0.2 https://appstore.home.mendix.com/link/app/47781/
